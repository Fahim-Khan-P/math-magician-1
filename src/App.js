import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Calculator from './components/calculator';
import Home from './components/home';
import NavBar from './components/nav';
import Quote from './components/quote';

function App() {
  return (
    <BrowserRouter>
      <NavBar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quote" element={<Quote />} />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
