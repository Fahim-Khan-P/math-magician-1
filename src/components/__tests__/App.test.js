import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../../App';

test('Test App rendering', () => {
  const tree = renderer
  .create(
      <App />
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});