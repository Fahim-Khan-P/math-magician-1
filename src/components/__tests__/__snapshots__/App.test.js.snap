// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App rendering 1`] = `
Array [
  <nav
    className="nav-bar"
  >
    <h1
      className="math-title"
    >
      Math Magician
    </h1>
    <div
      className="tags"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      |
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      |
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <div>
    <h2>
      Welcome to our Calculator
    </h2>
    <p
      className="info"
    >
      A calculator is a device that performs arithmetic operations on numbers. Basic calculators can do only addition, subtraction, multiplication and division mathematical calculations. However, more sophisticated calculators can handle exponential operations, square roots, logarithms, trigonometric functions and hyperbolic functions. Internally, some calculators perform all these functions by repeated addition processes.
      <br />
      OR
      <br />
      A calculator is a machine which allows people to do math operations more easily. For example, most calculators will add, subtract, multiply, and divide. There are simple basic calculators, scientific calculators & special purpose calculators (financial calculators etc.). Some also do square roots, and more complex calculators can help with calculus and draw function graphs.
    </p>
  </div>,
]
`;
